orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -634,50 +634,70 @@ void TrainingRunner::RunWithUpdate(VectorString& feed_names,
+
+    Status status = Status::OK();
+    pipeline_worker_pool_.workers[worker_id] = std::thread([&](
+        const size_t worker_id, const size_t step) {
+#ifdef ENABLE_NVTX_PROFILE
+      // Store the tag for the thread which runs session_.Run(...).
+      // It will be used to name range in Nvidia's visual profiler.
+      auto& profile_context = profile::Context::GetInstance();
+      profile_context.SetThreadTag(
+        std::this_thread::get_id(), std::to_string(step));
+#else
+      ORT_UNUSED_PARAMETER(step);
+#endif
+      status = session_.Run(
+        RunOptions(),
+        pipeline_worker_pool_.worker_states[worker_id].feed_names,
+        pipeline_worker_pool_.worker_states[worker_id].feeds,
+        pipeline_worker_pool_.worker_states[worker_id].fetch_names,
+        &(pipeline_worker_pool_.worker_states[worker_id].fetches));
+    }, worker_id, step_);
+
+    // Wait all workers to finish this round of pipeline parallelism.
+    // The last batch in a pipeline collects gradient and update the model.
+    // We must join here because main thread needs to access thread-produced
+    // fetches and those fetches must be ready.
+    pipeline_worker_pool_.JoinAll();
+
+    // If the updating thread fails, we fail with its error status.
+    ORT_ENFORCE(status.IsOK(), status.ErrorMessage());
+
+    // Copy back from thread-specific buffer to main thread's memory.
+    fetches = pipeline_worker_pool_.worker_states[worker_id].fetches;
+  } else {
+    // Pipeline is not enabled, so we run session using the main thread.
+
 #ifdef ENABLE_NVTX_PROFILE
     // Store the tag for the thread which runs session_.Run(...).
     // It will be used to name range in Nvidia's visual profiler.
     auto& profile_context = profile::Context::GetInstance();
     profile_context.SetThreadTag(
-      std::this_thread::get_id(), std::to_string(step));
-#else
-    ORT_UNUSED_PARAMETER(step);
+      std::this_thread::get_id(), std::to_string(step_));
 #endif
-    status = session_.Run(
-      RunOptions(),
-      pipeline_worker_pool_.worker_states[worker_id].feed_names,
-      pipeline_worker_pool_.worker_states[worker_id].feeds,
-      pipeline_worker_pool_.worker_states[worker_id].fetch_names,
-      &(pipeline_worker_pool_.worker_states[worker_id].fetches));
-  }, worker_id, step_);
-
-  // Wait all workers to finish this round of pipeline parallelism.
-  // The last batch in a pipeline collects gradient and update the model.
-  // We must join here because main thread needs to access thread-produced
-  // fetches and those fetches must be ready.
-  pipeline_worker_pool_.JoinAll();
-
-  // If the updating thread fails, we return with its error status.
-  ORT_THROW_IF_ERROR(status);
 
-  // Copy back from thread-specific buffer to main thread's memory.
-  fetches = pipeline_worker_pool_.worker_states[worker_id].fetches;
+    auto status = session_.Run(
+      RunOptions(),
+      feed_names,
+      feeds,
+      fetch_names,
+      &fetches);
```

Comment:
Nit: Can you please clang-format this file? I don't think the spacing is quite right here, but I'd prefer to leave that judgement to a tool :).

Reply:
Ok.

---
In reply to: [441847697](https://github.com/microsoft/onnxruntime/pull/4265#discussion_r441847697) [](ancestors = 441847697)